### DISCUSSION TITLE
Sorting Deta Base items by ULID key

### DISCUSSION THREAD

**Discussion by theseyan** [deta/cloud-docs#239] [General] [answered]
If I use time-sortable ULIDs as keys in Deta Base, will be able to get all items before or after a specific time by using the greater than/less than operator queries? Does Deta Base support such string comparisons?

**Comment by abdelhai** [comment, marked as answer]
yes, check our queries page: https://docs.deta.sh/docs/base/queries
and at the bottom where we have an example and list the limitations: https://docs.deta.sh/docs/base/queries#querying-keys

**Reply by theseyan** [reply]
Got it, thanks!

**Comment by Sokhavuth** [comment]
We can sort records by any column If we pull all records from the database first and sort them using vanilla JavaScript sort() method for sorting array of objects.
```js
let result = await req.mydb.posts.fetch()
let allItems = result.items

while(result.last){
         result = await req.mydb.posts.fetch({}, {last: result.last})
         allItems = allItems.concat(result.items)
   }

allItems.sort((a, b) => {     
            let da = new Date(a.date)
            let db = new Date(b.date)
            return db - da
})


const posts = allItems.slice(0, 9)
```
